File: Data Structures & Algorithms/anagram-groups/submission-17.cpp
#include <algorithm>

class Solution {
public:
    string hash(string &word) {
        int count[26] = {0};
        string result = "";
        for (char &c: word) {
            count[c - 97]++;
        }
        for (int counter: count) {
            result += std::to_string(counter) + ',';
        }
        return result;
    }

    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<string, vector<string>> counter;
        std::vector<std::vector<std::string>> results;

        for (string &word: strs) {
            string hashed = hash(word);
            counter[hashed].push_back(word);
        }

        for (auto pair: counter) {
            results.push_back(pair.second);
        }

        return results;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-17.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:

        from collections import Counter

        count = Counter(nums)
        result = [[] for _ in range(len(nums))]

        for key, value in count.items():
            result[value - 1].append(key)
        
        op = []
        for i in range(len(result) - 1, -1, -1):
            for num in result[i]:
                op.append(num)
                if len(op) == k:
                    return op
        
        return op

        